installer/datafiles/base_omsagent.data

```diff
@@ -211,6 +224,9 @@ if ${{PERFORMING_UPGRADE_NOT}}; then
    rmdir /etc/opt/microsoft/omsagent 2> /dev/null
    rmdir /etc/opt/microsoft 2> /dev/null
    rmdir /etc/opt 2> /dev/null
+elif ls ${{OMSHOME}}/ruby/lib/ruby/gems/2.2.0/cache/mysql2-0.4.4.gem 1> /dev/null 2> /dev/null; then
```

Comment:
It's not clear to me why you're doing this. The 'if' here (at line 212) is: If we're NOT performing an upgrade. So you're saying: If we're NOT performing an upgrade, then remove mysql2 gem.

You probably are thinking of reinstalling it later. BUT: this is problematic, as different packaging systems (RPM vs. DEB) have different rules for how/when postinstall is executed. That's why, when we do cleanup for before installation, we tend to do them in preinstall, as this is a safe consistent way to know when the code will be executed.

I suggest either:

1. Don't remove at all. Just modify install code to install if it hasn't already been installed (preferred, faster and less work), or
2. Relocate this to a preinstall script, where you can depend on proper execution ordering.

Also, if you really want to do this: That's kind of a weird way to determine if a file exists. Yeah, it works, but launching a program when the shell can do this natively?

How about something like:

```
elif [ -f ${{OMSHOME}}/ruby/lib/ruby/gems/2.2.0/cache/mysql2-0.4.4.gem ]; then
```

I think that's better, faster, but most importantly: clearer.